File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/map/submodelelement/DataElement.java
package org.eclipse.basyx.submodel.metamodel.map.submodelelement;

import org.eclipse.basyx.submodel.metamodel.api.submodelelement.IDataElement;
import org.eclipse.basyx.submodel.metamodel.map.modeltype.ModelType;

public class DataElement extends SubmodelElement implements IDataElement {
	public static final String MODELTYPE = "DataElement";

	public DataElement() {
		// Add model type
		putAll(new ModelType(MODELTYPE));
	}
}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/map/submodelelement/DataElement.cpp
/*
 * DataElement.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/map/submodelelement/DataElement.h>

namespace basyx {
namespace submodel {

DataElement::DataElement() :
  vab::ElementMap{},
  ModelType{IDataElement::Path::ModelType}
{}

DataElement::DataElement(basyx::object object) :
  vab::ElementMap{object},
	ModelType{ IDataElement::Path::ModelType }
{}

DataElement::DataElement(const IDataElement & other) :
  SubmodelElement{other},
	ModelType{ IDataElement::Path::ModelType }
{}

}
}
